crates/extensions/c8y_log_manager/src/actor.rs

```diff
@@ -469,3 +412,283 @@ impl TryIntoOperationStatusMessage for LogfileRequest {
+
+        log_file.write_all(data.as_bytes()).unwrap();
+
+        let line_counter = 0;
+        let max_lines = 4;
+        let filter_text = None;
+
+        let (line_counter, result) = LogManagerActor::read_log_content(
+            Path::new(file_path),
+            line_counter,
+            max_lines,
+            &filter_text,
+        )
+        .unwrap();
+
+        assert_eq!(line_counter, max_lines);
+        assert_eq!(result, "filename: file_a\nthis is the second line.\nthis is the third line.\nthis is the forth line.\nthis is the fifth line.\n");
+    }
+
+    #[test]
+    /// Inserting 5 lines of logs for each log file { file_a, ..., file_d }.
+    /// Each line contains the text: "this is the { line_number } line of { file_name }
+    /// where line_number { first, second, third, forth, fifth }
+    /// where file_name { file_a, ..., file_d }
+    ///
+    /// Requesting logs for log_type = "type_one", that are older than:
+    /// timestamp: 1970/01/01 00:00:03
+    ///
+    /// These are:
+    /// file_b and file_d
+    ///
+    /// file_d is the newest file, so its logs are read first. then file_b.
+    ///
+    /// Because only 7 lines are requested (and each file has 5 lines), the expedted
+    /// result is:
+    ///
+    /// - all logs from file_d (5)
+    /// - last two logs from file_b (2)
+    fn test_read_log_content_multiple_files() {
+        let (tempdir, logs_config) = prepare().unwrap();
+        let tempdir_path = tempdir.path().to_str().unwrap();
+
+        for (file_name, m_time) in [
+            ("file_a", 2),
+            ("file_b", 3),
+            ("file_c", 11),
+            ("file_d", 100),
+        ] {
+            let file_path = &format!("{tempdir_path}/{file_name}");
+
+            let mut log_file = std::fs::OpenOptions::new()
+                .append(true)
+                .create(false)
+                .write(true)
+                .open(file_path)
+                .unwrap();
+
+            let data = &format!("this is the first line of {file_name}.\nthis is the second line of {file_name}.\nthis is the third line of {file_name}.\nthis is the forth line of {file_name}.\nthis is the fifth line of {file_name}.");
+
+            log_file.write_all(data.as_bytes()).unwrap();
+
+            let new_mtime = FileTime::from_unix_time(m_time, 0);
+            set_file_mtime(file_path, new_mtime).unwrap();
+        }
+
+        let smartrest_obj = build_smartrest_log_request_object("type_one".to_string(), None, 7);
+        let mut actor = new_log_manager_actor(tempdir.path(), logs_config);
+
+        let result = actor.new_read_logs(&smartrest_obj).unwrap();
+        assert_eq!(result, String::from("filename: file_d\nthis is the first line of file_d.\nthis is the second line of file_d.\nthis is the third line of file_d.\nthis is the forth line of file_d.\nthis is the fifth line of file_d.\nfilename: file_b\nthis is the forth line of file_b.\nthis is the fifth line of file_b.\n"))
+    }
+}
```

Comment:
It'd be good to add a few unit tests that really test the actor interactions with its peer message boxes.

Reply:
See:

* test init messages: https://github.com/thin-edge/thin-edge.io/pull/1820/commits/f82f6362b6b609ae2b3fd0795b08bc67abdc869e
* test upload: https://github.com/thin-edge/thin-edge.io/pull/1820/commits/a864cb4c8a5affdba5885c609e1c1d1d1668994a
* fix of missing message highlighted by the test on init messages: https://github.com/thin-edge/thin-edge.io/pull/1820/commits/a3fda15953cd8b93b13587ac65b72b25b7390e29